### ISSUE TITLE
Gradle not resolving after compiling rainbow hat  

### ISSUE THREAD

**Issue by soorajshankar** [androidthings/contrib-drivers#85]
Error:Execution failed for task ':app:preDebugAndroidTestBuild'.
> Conflict with dependency 'com.android.support:support-annotations' in project ':app'. Resolved versions for app (24.2.0) and test app (25.4.0) differ. See https://d.android.com/r/tools/test-apk-dependency-conflicts.html for details.

**Comment by Fleker** [comment]
Can you post the line in your gradle file you're using to import the
rainbow hat?

**Comment by soorajshankar** [comment]
I fixed that by removing expresso from dependency .
But seems some version mismatching is there.
BTW this was the way I imported rainbowhat
>     compile 'com.google.android.things.contrib:driver-rainbowhat:+'
also tried with older version 0.7 and all 

> Min SDK version 26
> target SDK version 26 (tried with 24 as min also then things was not compatible)

**Comment by Fleker** [comment]
Looks like Rainbow Hat uses [24.2.0 of Android Support Annotations](https://github.com/androidthings/contrib-drivers/blob/master/rainbowhat/build.gradle#L32), but only for testing. I don't know why it would conflict with your version.

**Comment by soorajshankar** [comment]
I think Android Support Annotations are defined as **Implementation** but not **Test Implementation**

I doubt that could be a problem if I use another support library with different version

**Comment by Fleker** [comment]
Yes you're right it isn't defined as **testImplementation**, but I do wonder why this library is using it. It probably should be updated or perhaps removed.

But **implementation** shouldn't pass the dependencies through AFAIK.
